web/src/pages/api/post/getPost.ts

```diff
@@ -0,0 +1,50 @@
+import { NextApiRequest, NextApiResponse } from "next";
+
+import { prisma } from "../../../server/db/client";
+import { validateRequest } from "../../../utils/jwt";
+
+interface Request extends NextApiRequest {
+  body: {
+    postId: string;
+  };
+}
+
+export default async function get(req: Request, res: NextApiResponse) {
+  // Validate if the user has a valid JWT token
+  if (!(await validateRequest(req))) {
+    return res.status(401).json({ error: "User not logged in." });
+  }
+
+  const { postId } = req.body;
```

Comment:
This is probably my bad for introducing this pattern but I later realized that GET endpoints cannot have a body. You can only use query params

Reply:
So technically there is nothing which prevents this, but it is a violation of the spec. Some people do send body over GET though

Reply:
Yea, I even saw an API that requires you to send a body in a get in a forum where people were discussing this lol, but it is not part of the standard so better to avoid it

Reply:
The issue is that query params have a length limit that is reachable in practice, so what do you do then?

Reply:
For us though that shouldn't be an issue

Reply:
According to the spec there is no limit https://stackoverflow.com/questions/812925/what-is-the-maximum-possible-length-of-a-query-string

But seems like the limit is implemented on a per browser level

Reply:
Regardless, I dont think we'll run into this character limit

My main gripe with not using query params for get is that the native fetch function wont let you send GET requests with a body unless there is a workaround but even then it'll feel hacky.